### ISSUE TITLE
Logging Numpy arrays, Torch Tensors and Tensorflow Tensors

### ISSUE THREAD

**Issue by BartKeulen** [rlworkgroup/dowel#46]
Hi,

Thank you for this nice a simple tool for logging machine learning research. I often encounter situations where I would like to save multi-dimensional Numpy arrays. For example, the observation at each time-step in a reinforcement learning experiment.

It would be nice to have an output logger that supports Numpy arrays, Pytorch Tensors and Tensorflow Tensors.

I have written a simple output logger, [`NpzOutput`](https://github.com/BartKeulen/dowel/blob/npz_output/src/dowel/npz_output.py), that writes Numpy arrays to a `.npz` file using Numpys `savez` functions. It is not optimal (no incremental saving), but thought I share it in case somebody is interested.

**Comment by ryanjulian** [comment]
@BartKeulen thanks for sharing! We'd love a pull request, if you have the time.

**Comment by BartKeulen** [comment]
Yes, that is possible. I will write the tests somewhere in the next 2 to 3 weeks and submit a pull request.
